src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -617,49 +635,69 @@ impl<'index, D: Decoder> IndexReader<'index, D> {
         // Check if the current buffer is empty. The GC might clean out a block so we have to
         // continue checking until we find a block with data.
         while self.current_buffer.fill_buf()?.is_empty() {
-            if self.current_block_idx + 1 >= self.blocks.len() {
+            if self.current_block_idx + 1 >= self.ii.blocks.len() {
                 // No more blocks to read from
                 return Ok(None);
             };
 
             self.set_current_block(self.current_block_idx + 1);
         }
 
-        let base = D::base_id(self.current_block, self.last_doc_id);
+        let base = D::base_id(&self.ii.blocks[self.current_block_idx], self.last_doc_id);
         let result = self.decoder.decode(&mut self.current_buffer, base)?;
 
         self.last_doc_id = result.doc_id;
 
         Ok(Some(result))
     }
 
+    /// Seek to the first record whose ID is higher or equal to the given document ID. If the end
+    /// of the index is reached before finding the document ID, then `None` is returned.
+    pub fn seek_record(
```

Comment:
Something seems wrong here as my iterator tests fail when using it. I'm investigating.